### ISSUE TITLE
Update Controls?

### ISSUE THREAD

**Issue by ttsgeb** [LaboratoryB/camfive#10]
Our good friend, @ttsgeb has made a repo, https://github.com/chattlab-makerspace/Laser-Automation, that has fun stuff for making it so the laser controls fans and the chiller and whatnot more intelligently.  This depends on a Ruida controller such as https://www.ruidacontroller.com/rdv6442g/, which is likely a great upgrade for the laser.

**Comment by DaBlender** [comment]
Definitely. Does it make sense to merge this with software stack?  I posted updates there after exhausting current controller progress.

**Comment by ttsgeb** [comment]
I think we should wait until we've confirmed whether we actively need to upgrade the controller to use a modern software stack.  IMO, if that's the case, the software stack issue should be merged into this one, as the controller then becomes the primary concern.

**Comment by DaBlender** [comment]
It is confirmed. The compatibility with cam 5 needs a 5th gen controller.
Plus the cost to get a dongle approaches the value of the best possible
card. Unless someone is debugging the dongle check, that in itself is high
paying work that again exceeds worth of new board and stuck with garbage
software. I’m ok if not bought now, I’ll use my small laser until I can
afford it, it’s going to be the first upgrade I get at all as it increases
productivity immensely. A few jobs it pays for itself. Plus other add ons.
Better nozzles. All upgrades will be in eventually. I’m aware it’s another
cost. I’m having an estate sale of high value comes I’m ordering it
immediately as it takes time to arrive and this needs to work soon.
Opportunity loss is huge right now. It can produce thousands in value
product that sells per day. This frees me from in person and can wholesale
to most art tourist spots.

**Comment by ttsgeb** [comment]
https://www.aliexpress.us/item/2251832528159197.html
Okay, so honestly? At ~$300, this is entirely worth the expense.  It adds modern 3rd party software support and a huge community of people that use this controller.  Cloudrays is also a decent brand that puts some amount of effort into maintaining a good image, so they're likely selling what they say they are.  Seeing the game being played, I believe this is a substantially better use of time and money than trying to make the existing controller behave.  This is my actually professional opinion as someone that has commissioned several lasers and works with them on a daily basis.

It is worth noting that the price does not include the software I prefer, lightburn, which is expected to go up to $199 on the 30th, however there should be a usable driver software included and lightburn will let you do 30 days of free trial.

**Comment by ttsgeb** [comment]
![image](https://github.com/user-attachments/assets/5e53d5ec-f802-4c18-97ee-23141c570032)
I've tried to sus out what a few of the wires are to help speed along switching controllers.
Everything with an LED next to it is an input of some flavor, you could probably get pretty far by turning the machine on and then activating switches to see what turns on.
It's using prox switches for the X and Y limits, and looks to have a Z min and max as standard microswitches.  U is likely a secondary Z axis (are there two motors for Z?)

The unlabeled wires on the right I suspect are related to machine power and turning on the accessories.  The left might be an emergency stop? I'm not sure, there are a lot of options and I don't have hands on the thing.  Shouldn't be too tough to figure out, though.

**Comment by DaBlender** [comment]
I like the light object .com one more.

**Comment by DaBlender** [comment]
If I can find my no contact continuity tester I don’t know what’s quicker
besides knowing the current setup

**Comment by ttsgeb** [comment]
What specifically do you like more about the option that's $100 more expensive?
If it's the additional buttons, my experience is that this already has buttons you never care to use, and there is no need for more.  If you're instead thinking of the touch-screen model, I simply don't like touch screens.

**Comment by DaBlender** [comment]
Reliable source from community types that recommended the company. Then I
got live support and they helped select it, cool questions. It did match
another users model.

I went on reputation and service was great. They are quick. I did not
attempt to do that with another model. You can just say it costs more
clearly, that wasn’t a factor for me either I just now looked at the button
riddle and I yes would press them all. I didn’t look at a warranty but
maybe they do. Definitely heard.

**Comment by DaBlender** [comment]
Can we move this to critical?  We can’t use it without a dongle or new controller

**Comment by heaventwig** [comment]
LightObject DSP controller on order, expected delivery time is three business days. If they ship tomorrow, or even Tuesday, it should get here this week.

**Comment by heaventwig** [comment]
LightObject has created their UPS label. UPS says they don't yet have the package at this time.

**Comment by DaBlender** [comment]
Good catch on U axis. We were thinking rotary capacity but will trace. Ty.

**Comment by ttsgeb** [comment]
I am curious how well my interpretation of the silkscreen and wires lined up with what Google Lens had to say.

**Comment by heaventwig** [comment]
UPS says this is arriving tomorrow.
![image](https://github.com/user-attachments/assets/2f9cfeb3-8a5a-417b-b950-23913e042fb8)

**Comment by heaventwig** [comment]
They're heeeere... (the DSP parts).
![image](https://github.com/user-attachments/assets/3908916b-8f7b-44de-ab0b-ee4c2e707077)

**Comment by heaventwig** [comment]
And these connections look a heck of a lot simpler than the old board.
![image](https://github.com/user-attachments/assets/9581fec9-97bd-41b3-ae4a-fb36fb412486)
And there's even a little baggie of these connectors.
![image](https://github.com/user-attachments/assets/094eb3e5-b5e0-4a45-aa94-d5c3be8e01ae)

But, like, what in the what are these USB-A to USB-A cables for, that's ... big yikes.
![image](https://github.com/user-attachments/assets/885700eb-23c7-4cd4-98c0-b496521685fc)
![image](https://github.com/user-attachments/assets/fb29a05a-7d2a-4830-be6c-012936e8068a)

**Comment by DaBlender** [comment]
I think that’s for panel mounting usb extension to outside case. Prob for thumb drive?  Glad we have connections!  Ty!  Edit oh whoops I see it. Yikes! Random guess control panel

**Comment by ttsgeb** [comment]
![image](https://github.com/user-attachments/assets/6e98aebd-d704-4157-897f-325dad0d90f2)

Basically, the A male to A male is because they made a bad decision in the kit on extending the Type B comms port on the controller.  Personally, I would just run a long Type B into the machine, but you could also look for a panel mount Type B extender, or just ignore it and put the thing on the network (which works great with LightBurn)

**Comment by ttsgeb** [comment]
Honestly, both types of connectors definitely have their place. You're still going to want the ferrules on the wires, but it's going to be harder to swap out individual wires now because you're losing the spring release in exchange for screw cages.  It's not really a big deal, but in some ways that could be seen as a downgrade.

Proper labeling and being able to remove wires as groups is a huge advantage, though. Not to mention support from actively developed software.
